hal/include/bmp180_device.h

```diff
@@ -0,0 +1,179 @@
+
+	SerialPort serial_;
+	Bmp180Protocol protocol_;
+
+	uint8_t default_i2c_addr_ = 0xEE;
+
+	static const uint8_t kBUFFER_SIZE = 32;
+
+	// CONSTANTS
+	static const uint8_t kINITIALIZE_OVERSAMP_SETTING = 0;
+	static const uint8_t kINITIALIZE_SW_OVERSAMP = 0;
+	static const uint8_t kGEN_READ_WRITE_DATA_LENGTH = 1;
+	static const uint8_t kTEMPERATURE_DATA_LENGTH = 2;
+	static const uint8_t kPRESSURE_DATA_LENGTH = 3;
+	static const uint8_t kSW_OVERSAMP = 1;
+	static const uint8_t kOVERSAMP_SETTING = 3;
+	static const uint8_t k2MS_DELAY = 2;
+	static const uint8_t k3MS_DELAY = 3;
+	static const uint8_t kAVERAGE = 3;
+	static const uint8_t kCHECK_DIVISOR = 0;
+	static const uint8_t kDATA_MEASURE = 3;
+	static const uint8_t kCALCULATE_TRUE_PRESSURE = 8;
+	static const uint8_t kCALCULATE_TRUE_TEMPERATURE = 8;
+
+	enum {
+		kSHIFT_BIT_POSITION_BY_01_BITS = 1,
+		kSHIFT_BIT_POSITION_BY_02_BITS = 2,
+		kSHIFT_BIT_POSITION_BY_04_BITS = 4,
+		kSHIFT_BIT_POSITION_BY_06_BITS = 6,
+		kSHIFT_BIT_POSITION_BY_08_BITS = 8,
+		kSHIFT_BIT_POSITION_BY_11_BITS = 11,
+		kSHIFT_BIT_POSITION_BY_12_BITS = 12,
+		kSHIFT_BIT_POSITION_BY_13_BITS = 13,
+		kSHIFT_BIT_POSITION_BY_15_BITS = 15,
+		kSHIFT_BIT_POSITION_BY_16_BITS = 16,
+	};
+
+	// register definitions
+	static const int8_t kPROM_START_ADDR = 0xAA;
+	static const int8_t kPROM_DATA_LEN = 22;
+
+	static const int8_t kCHIP_ID_REG = 0xD0;
+	static const int8_t kVERSION_REG = 0xD1;
+
+	static const int8_t kCTRL_MEAS_REG = 0xF4;
+	static const int8_t kADC_OUT_MSB_REG = 0xF6;
+	static const int8_t kADC_OUT_LSB_REG = 0xF7;
+
+	static const int8_t kSOFT_RESET_REG = 0xE0;
+
+	// temperature measurement
+	static const int8_t kT_MEASURE = 0x2E;
+	// pressure measurement
+	static const int8_t  kP_MEASURE = 0x34;
+
+	// TO be spec'd by GL or SB
+	static const uint8_t  kTEMP_CONVERSION_TIME = 5;
+
+	static const int16_t kPARAM_MG = 3038;
+	static const int16_t kPARAM_MH = -7357;
+	static const int16_t kPARAM_MI = 3791;
+
+	// ARRAY SIZE DEFINITIONS
+	static const uint8_t kTEMPERATURE_DATA_BYTES = 2;
+	enum {
+		kTEMPERATURE_MSB_DATA,
+		kTEMPERATURE_LSB_DATA
+	};
+
+	static const uint8_t kPRESSURE_DATA_BYTES = 3;
+	enum {
+		kPRESSURE_MSB_DATA,
+		kPRESSURE_LSB_DATA,
+		kPRESSURE_XLSB_DATA,
+	};
+
+	static const uint8_t kCALIB_DATA_SIZE = 22;
+	enum {
+		kCALIB_PARAM_AC1_MSB,
+		kCALIB_PARAM_AC1_LSB,
+		kCALIB_PARAM_AC2_MSB,
+		kCALIB_PARAM_AC2_LSB,
+		kCALIB_PARAM_AC3_MSB,
+		kCALIB_PARAM_AC3_LSB,
+		kCALIB_PARAM_AC4_MSB,
+		kCALIB_PARAM_AC4_LSB,
+		kCALIB_PARAM_AC5_MSB,
+		kCALIB_PARAM_AC5_LSB,
+		kCALIB_PARAM_AC6_MSB,
+		kCALIB_PARAM_AC6_LSB,
+		kCALIB_PARAM_B1_MSB,
+		kCALIB_PARAM_B1_LSB,
+		kCALIB_PARAM_B2_MSB,
+		kCALIB_PARAM_B2_LSB,
+		kCALIB_PARAM_MB_MSB,
+		kCALIB_PARAM_MB_LSB,
+		kCALIB_PARAM_MC_MSB,
+		kCALIB_PARAM_MC_LSB,
+		kCALIB_PARAM_MD_MSB,
+		kCALIB_PARAM_MD_LSB,
+	};
+
+	// brief This structure holds all device specific calibration parameters
+	struct calib_param_t
+	{
+		int ac1; 			// <calibration ac1 data
```

Comment:
If these parameters are size-related, consider using `int32_t` or `uint32_t`. If not, just ignore this comment.